### ISSUE TITLE
Intermittent federation with other instances

### ISSUE THREAD

**Issue by vmatekole** [tootsuite/mastodon#5593]
We recently bumped to version 2.0 on social.coop. Since then we had a number of issues, all resolved, apart from federation with other instances and ability to remote follow, here are the symptoms suffered:

1. For example my account — https://social.coop/web/accounts/3298 attempts to follow -> https://social.coop/web/accounts/11809, who is on mastodon.sdf.org appears to follow successfully 
but later on return I get a status of Awaiting for Approval (hourglass).

2. A user reported they can only see a subset of federated instances for example we can see GNU Social but not mastodon.social. 

We don't see anything in the logs that point to an issue... Please can you advise how we could troubleshoot/reset federation with others? 

* * * *

- [ ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by mayel** [comment]
Just to note that we're having difficulty pinning down this federation issue at [social.coop](http://social.coop) as the 2.0 version upgrade was done at the same time as a total infrastructure migration (from one server to two, now using docker swarm).
Really hoping for some help from the community now if anybody has an idea!

**Comment by nightpool** [comment]
Hard to diagnose this kind of issue without logs from the other side. However, I've often seen ipv6 problems cause this kind of behavior, especially with new infrastructure. Are you advertising a AAAA record? If you ping that record, does it work? What about curl -6 from an ipv6 enabled system?

**Comment by mayel** [comment]
Thanks @nightpool, we didn't have a AAAA record, I've now I added it but curl -6 times out.

**Comment by nightpool** [comment]
Well, if you DON'T have an AAAA record, then you're probably *better* off.
The problem is when you DO have a AAAA record but it doesn't work.

**Comment by mayel** [comment]
Right, so that probably wasn't the issue. I've the AAAA for now then.

**Comment by nightpool** [comment]
These are the errors I'm getting on cybre.space for social.coop:

```
 WARN: HTTP::StateError: Subscribe failed for jjg@social.coop: can't follow 301 Moved Permanently redirect on http://social.coop/api/push
```

```
Nov  6 09:22:44 neo bundle[6425]: 2017-11-06T08:22:44.112Z 6425 TID-os80lkmuc WARN: {"context":"Job raised exception","job":{"class":"Pubsubhubbub::SubscribeWorker","args":[34831],"retry":10,"queue":"push","unique":"until_executed","dead":false,"jid":"9d4dd026bfa2c77aa901d0cb","created_at":1509947520.3794239,"enqueued_at":1509956563.8902073,"unique_prefix":"uniquejobs","unique_args":[34831],"unique_digest":"uniquejobs:a906985642059313373a3d9b56734274","error_message":"Subscribe failed for lehighcommunalist@social.coop: can't follow 301 Moved Permanently redirect on http://social.coop/api/push","error_class":"HTTP::StateError","failed_at":1509947548.4432755,"retry_count":2,"retried_at":1509956564.077466},"jobstr":"{\"class\":\"Pubsubhubbub::SubscribeWorker\",\"args\":[34831],\"retry\":10,\"queue\":\"push\",\"unique\":\"until_executed\",\"dead\":false,\"jid\":\"9d4dd026bfa2c77aa901d0cb\",\"created_at\":1509947520.3794239,\"enqueued_at\":1509956563.8902073,\"unique_prefix\":\"uniquejobs\",\"unique_args\":[34831],\"unique_digest\":\"uniquejobs:a906985642059313373a3d9b56734274\",\"error_message\":\"Subscribe failed for lehighcommunalist@social.coop: can't follow 301 Moved Permanently redirect on http://social.coop/api/push\",\"error_class\":\"HTTP::StateError\",\"failed_at\":1509947548.4432755,\"retry_count\":1,\"retried_at\":1509949357.715107}"}
Nov  6 09:22:44 neo bundle[6425]: 2017-11-06T08:22:44.116Z 6425 TID-os80lkmuc WARN: HTTP::StateError: Subscribe failed for lehighcommunalist@social.coop: can't follow 301 Moved Permanently redirect on http://social.coop/api/push
Nov  6 09:22:44 neo bundle[6425]: 2017-11-06T08:22:44.116Z 6425 TID-os80lkmuc WARN: /home/mastodon/live/vendor/bundle/ruby/2.4.0/gems/http-2.2.2/lib/http/redirector.rb:90:in `redirect_to'
```

do you have HTTP redirects set up on your server in some way?

**Comment by nightpool** [comment]
Did you happen to change LOCAL_DOMAIN or WEB_DOMAIN?

**Comment by vmatekole** [comment]
@nightpool does federation ONLY happen on plain-text HTTP? Currently, we redirect all HTTP requests to SSL.

**Comment by vmatekole** [comment]
@nightpool I have disabled redirects for all requests on /api/push. Let me know if you still suffer from issues.

**Comment by vmatekole** [comment]
Additionally, we discovered that LOCAL_HTTPS was still set to false. This has been set to TRUE now.

**Comment by vmatekole** [comment]
@nightpool setting LOCAL_HTTPS wasn't intentional, was just missed during the migration. We also disabled redirects for /api/push too. Federation is working once again. Thanks for your insights, appreciated! Please close.
